middlewares/cors.go

```diff
@@ -0,0 +1,179 @@
+package middlewares
+
+import (
+	"github.com/savsgio/atreugo/v10"
+	"github.com/savsgio/go-logger"
+
+	"strconv"
+	"strings"
+)
+
+type CorsOptions struct {
+	AllowedOrigins   []string
+	AllowedHeaders   []string
+	AllowMaxAge      int
+	AllowedMethods   []string
+	ExposedHeaders   []string
+	AllowCredentials bool
+}
+
+type CorsHandler struct {
+	maxAge            int
+	allowCredentials  bool
+	allowedOriginsAll bool
```

Comment:
It's not necessary because you could check if value is `*` in `isAllowedOrigin()`